c-data-interface.ipynb

```diff
@@ -0,0 +1,261 @@
+    "\n",
+    "While there are specifications to share Arrow data between processes or over the network (e.g. the IPC messages), the **Arrow C Data Interface** is meant to actually zero-copy share the data between different libraries *within the same process* (i.e. actually share the same buffers in memory).\n",
+    "\n",
+    "The Arrow C Data Interface defines a set of small C structures\n",
+    "\n",
+    "```c\n",
+    "struct ArrowSchema {\n",
+    "  const char* format;\n",
+    "  const char* name;\n",
+    "  const char* metadata;\n",
+    "  int64_t flags;\n",
+    "  int64_t n_children;\n",
+    "  struct ArrowSchema** children;\n",
+    "  struct ArrowSchema* dictionary;\n",
+    "\n",
+    "  // Release callback\n",
+    "  void (*release)(struct ArrowSchema*);\n",
+    "  // Opaque producer-specific data\n",
+    "  void* private_data;\n",
+    "};\n",
+    "\n",
+    "struct ArrowArray {\n",
+    "  int64_t length;\n",
+    "  int64_t null_count;\n",
+    "  int64_t offset;\n",
+    "  int64_t n_buffers;\n",
+    "  int64_t n_children;\n",
+    "  const void** buffers;\n",
+    "  struct ArrowArray** children;\n",
+    "  struct ArrowArray* dictionary;\n",
+    "\n",
+    "  // Release callback\n",
+    "  void (*release)(struct ArrowArray*);\n",
+    "  // Opaque producer-specific data\n",
+    "  void* private_data;\n",
+    "};\n",
+    "```\n",
+    "\n",
+    "\n",
+    "The C Data Interface passes Arrow data buffers through memory pointers. So, by construction, it allows you to share data from one runtime to another without copying it. Since the data is in standard Arrow in-memory format, its layout is well-defined and unambiguous.\n",
+    "\n",
+    "And in the examples up to now, when we created a `nanoarrow.Array` from a `pyarrow` array (or vice versa), we were actually using the Arrow C Data Interface to share the data zero-copy under the hood (and you might recognize the structure members from the nanoarrow display we have been using to inspect our data). \n",
+    "Similarly, other libraries like polars, duckdb, datafusion, reticulate (connect R and python), ... are all leveraging the Arrow C Data Interface to interchange data zero-copy.\n",
+    "\n",
+    "### Arrow PyCapsule Interface\n",
+    "\n",
+    "While the Arrow C data interface specify how to share the data at the C(FFI) level, it doesn't specify how Python libraries should expose these structs to other libraries. Enter the [**Arrow PyCapsule Interface**](https://arrow.apache.org/docs/format/CDataInterface/PyCapsuleInterface.html), which standardizes the usage of `PyCapsule` objects to share those structs at the Python level and the protocol methods (`__arrow_c_array__` et al) to export those capsules.\n",
```

Comment:
Is `Enter the ...` in the beginning of the second sentence by accident? It is very possible that I do not understand the meaning